src/main/java/org/sagebionetworks/bridge/udd/synapse/SynapseDownloadFromTableParameters.java

```diff
@@ -0,0 +1,139 @@
+    public LocalDate getEndDate() {
+        return endDate;
+    }
+
+    /** Temp dir to download CSV and file handles to. */
+    public File getTempDir() {
+        return tempDir;
+    }
+
+    /** Schema key, used to determine file names. */
+    public UploadSchema getSchema() {
+        return schema;
+    }
+
+    /** Parameter class builder. */
+    public static class Builder {
+        private String synapseTableId;
+        private String healthCode;
+        private LocalDate startDate;
+        private LocalDate endDate;
+        private File tempDir;
+        private UploadSchema schema;
+
+        /** @see SynapseDownloadFromTableParameters#getSynapseTableId */
+        public Builder withSynapseTableId(String synapseTableId) {
+            this.synapseTableId = synapseTableId;
+            return this;
+        }
+
+        /** @see SynapseDownloadFromTableParameters#getHealthCode */
+        public Builder withHealthCode(String healthCode) {
+            this.healthCode = healthCode;
+            return this;
+        }
+
+        /** @see SynapseDownloadFromTableParameters#getStartDate */
+        public Builder withStartDate(LocalDate startDate) {
+            this.startDate = startDate;
+            return this;
+        }
+
+        /** @see SynapseDownloadFromTableParameters#getEndDate */
+        public Builder withEndDate(LocalDate endDate) {
+            this.endDate = endDate;
+            return this;
+        }
+
+        /** @see SynapseDownloadFromTableParameters#getTempDir */
+        public Builder withTempDir(File tempDir) {
+            this.tempDir = tempDir;
+            return this;
+        }
+
+        /** @see SynapseDownloadFromTableParameters#getSchema */
+        public Builder withSchema(UploadSchema schema) {
+            this.schema = schema;
+            return this;
+        }
+
+        /** Builds the parameters object and validates parameters. */
+        public SynapseDownloadFromTableParameters build() {
+            if (Strings.isNullOrEmpty(synapseTableId)) {
+                throw new IllegalStateException("synapseTableId must be specified");
+            }
+
+            if (Strings.isNullOrEmpty(healthCode)) {
+                throw new IllegalStateException("healthCode must be specified");
+            }
+
+            if (startDate == null) {
+                throw new IllegalStateException("startDate must be specified");
+            }
+
+            if (endDate == null) {
+                throw new IllegalStateException("endDate must be specified");
+            }
+
+            if (startDate.isAfter(endDate)) {
+                throw new IllegalStateException("startDate can't be after endDate");
+            }
+
+            if (tempDir == null) {
+                throw new IllegalStateException("tempDir must be specified");
+            }
+
+            if (schema == null) {
+                throw new IllegalStateException("schema must be specified");
+            }
```

Comment:
I feel like IllegalState captures the exception better, since this is because of the state of the Builder. That said, I can see it going either way.

After discussing with Alx offline, we decided that this is minor, and there's a lot of these, so we'll leave them as is.